content/intro-to-storybook/react-native/en/composite-component.md

```diff
@@ -18,303 +18,311 @@ Since `Task` data can be sent asynchronously, we **also** need a loading state t
+
+export default {
+  component: TaskList,
+  title: "TaskList",
+  decorators: [
+    (Story) => (
+      <View style={{ padding: 42, backgroundColor: "#26c6da", flex: 1 }}>
+        <Story />
+      </View>
+    ),
+  ],
+};
+
+export const Default = {
+  args: {
+    // Shaping the stories through args composition.
+    // The data was inherited from the Default story in Task.stories.js.
+    tasks: [
+      { ...TaskStory.args.task, id: "1", title: "Task 1" },
+      { ...TaskStory.args.task, id: "2", title: "Task 2" },
+      { ...TaskStory.args.task, id: "3", title: "Task 3" },
+      { ...TaskStory.args.task, id: "4", title: "Task 4" },
+      { ...TaskStory.args.task, id: "5", title: "Task 5" },
+      { ...TaskStory.args.task, id: "6", title: "Task 6" },
+    ],
+  },
+};
+
+export const WithPinnedTasks = {
+  args: {
+    // Shaping the stories through args composition.
+    // Inherited data coming from the Default story.
+    tasks: [
+      ...Default.args.tasks.slice(0, 5),
+      { id: "6", title: "Task 6 (pinned)", state: "TASK_PINNED" },
+    ],
+  },
+};
+
+export const Loading = {
+  args: {
+    tasks: [],
+    loading: true,
+  },
+};
+
+export const Empty = {
+  args: {
+    // Shaping the stories through args composition.
+    // Inherited data coming from the Loading story.
+    ...Loading.args,
+    loading: false,
+  },
+};
 ```
 
-As you may have noticed, the `addDecorator()` was used in the previous chapter and in this one, it allows us to add some “context” to the rendering of each task. In this case we add padding around the list to make it easier to visually verify.
+As you may have noticed, a decorator was used in the previous chapter and in this one, it allows us to add some “context” to the rendering of each task. In this case we add padding around the list to make it easier to visually verify.
 
 <div class="aside">
 <a href="https://storybook.js.org/docs/react/writing-stories/decorators"><b>Decorators</b></a> are a way to provide arbitrary wrappers to stories. In this case we’re using a decorator to add styling. They can also be used to wrap stories in “providers” –i.e. library components that set React context.
 </div>
 
 `task` supplies the shape of a `Task` that we created and exported from the `Task.stories.js` file. Similarly, `actions` defines the actions (mocked callbacks) that a `Task` component expects, which the `TaskList` also needs.
 
-Don't forget that this story also needs to be added to `storybook/index.js` so that it can be picked up and displayed.
-
-Change the `configure()` method to the following:
+React Native storybook uses a script to generate imports for story files due to how dynamic imports are not supported in React Native. Lets run that script so our new story shows up:
 
-```javascript
-// storybook/config.js
-configure(() => {
-  require('../components/Task.stories.js');
-  require('../components/TaskList.stories.js');
-}, module);
+```sh
+yarn sb-rn-get-stories
```

Comment:
@dannyhw tie in this snippet with some text / section similar to the "Simple Component" config section mentioning that with each new component/ story, the readers are required to run the command to allow Storybook to pull in the updated / new stories.

Reply:
I've done something similar to this